proc_trees:
  - name: far_buoys
    proc_units:
      - name: histogram
      - name: blur
        blur_type: 1
        kernel_size: 3
      - name: threshold
        threshold_type: 0
        thresh_value: 10
      - name: morphology
        kernel_size_x: 4
        kernel_size_y: 4
      - name: wall_remover
      - name: blob_detector
        target: 1
      - name: far_buoys_detector
        weight_goal: 20
  - name: buoys
    proc_units:
      - name: blur
        blur_type: 1
        kernel_size: 3
      - name: threshold
        threshold_type: 0
        thresh_value: 10
      - name: wall_remover
      - name: blob_detector
        target: 0
      - name: buoys_detector
        roi_needed: false
        weight_goal: 100
  - name: fence
    proc_units:
      - name: blur
        blur_type: 1
        kernel_size: 3
      - name: threshold
        threshold_type: 0
        thresh_value: 10
      - name: morphology
        kernel_size_x: 4
        kernel_size_y: 4
      - name: wall_remover
      - name: blob_detector
        target: 2
      - name: fence_detector
        weight_goal: 20
  - name: wall
    proc_units:
      - name: blur
        blur_type: 1
        kernel_size: 3
      - name: morphology
        kernel_size_x: 4
        kernel_size_y: 4
      - name: threshold
        threshold_type: 0
        thresh_value: 10

# Comment this line in order to have no default proc tree
default: "buoys"
